corehq/apps/consumption/shortcuts.py

```diff
@@ -85,30 +84,26 @@ def _update_or_create_default(domain, amount, default, type, **kwargs):
         default.save()
         return default
     else:
-        default = DefaultConsumption(domain=domain, default_consumption=amount, type=type, **kwargs)
+        default = SQLDefaultConsumption(domain=domain, default_consumption=amount, type=type, **kwargs)
         default.save()
         return default
 
 
-def hashable_key(key):
-    """
-    Convert the key from couch into something hasable.
-    Mostly, just need to make it a tuple and remove the special
-    {} value.
-    """
-    return tuple('{}' if item == {} else item for item in key)
-
-
 def build_consumption_dict(domain):
     """
-    Takes raw rows from couch and builds a dict to 
+    Takes raw rows from couch and builds a dict to
     look up consumption values from.
     """
-    raw_rows = get_domain_monthly_consumption_data(domain)
+    SQLDefaultConsumption.objects.filter(domain=domain)
 
     return dict(
-        (hashable_key(row['key']), Decimal(row['value']))
-        for row in raw_rows if row['value']
+        (tuple((
+            obj.domain,
+            obj.product_id,
+            obj.supply_point_type,
+            obj.supply_point_id,
+        )), obj.default_consumption)
+        for obj in SQLDefaultConsumption.objects.filter(domain=domain) if obj.default_consumption
     )
```

Comment:
At the risk of getting bogged down in details, I'll throw in my 5c: You could use a dictionary comprehension. And that `hashable_key()` function would make the comprehension more readable, so I'd advocate resurrecting it:

```suggestion
    return {
        hash_key(obj): obj.default_consumption
        for obj in SQLDefaultConsumption.objects.filter(domain=domain)
        if obj.default_consumption
    }
    
def hash_key(obj):
    return (
        obj.domain,
        obj.product_id,
        obj.supply_point_type,
        obj.supply_point_id,
    )
```